bftengine/src/bftengine/SequenceWithActiveWindow.hpp

```diff
@@ -55,6 +115,10 @@ class SequenceWithActiveWindow {
     return ((n >= beginningOfActiveWindow) && (n < (beginningOfActiveWindow + WindowSize)));
   }
 
+  bool IsPressentInHistory(NumbersType n) const {
```

Comment:
A bit misleading name since it checks both history and active window.
I would make a separate method for history.
Minor: starts with lower case.

Reply:
Yes, will separate the two. I think to simplify request in ReplicaImp.cpp I'll add a method that explicitly searches both.